src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -38,6 +38,37 @@ internal abstract partial class AbstractReferenceFinder : IReferenceFinder
         public abstract ValueTask<ImmutableArray<FinderLocation>> FindReferencesInDocumentAsync(
             ISymbol symbol, FindReferencesDocumentState state, FindReferencesSearchOptions options, CancellationToken cancellationToken);
 
+        protected virtual ValueTask<(bool matched, CandidateReason reason)> SymbolsMatchAsync(
```

Comment:
these are moves up from the static lambdas we had into normal methods.

Reply:
note: This is a virtual as there's one finder currently that overrides it (though i'd like to get rid of htat asap).

Reply:
because it's virtual, it became an instance method, so a lot of statics become instance methods